plugin/src/main/kotlin/org/gradle/github/dependency/extractor/internal/DependencyExtractorService.kt

```diff
@@ -160,10 +169,13 @@ abstract class DependencyExtractorService :
         }
     }
 
-    fun writeAndGetSnapshotFile(): File =
+    private fun writeAndGetSnapshotFile(): File =
         fileWriter.writeDependencyManifest(gitHubRepositorySnapshotBuilder.build())
 
     override fun close() {
+        if (thrownExceptions.isNotEmpty()) {
+            throw MultipleBuildFailures(thrownExceptions)
```

Comment:
It would be great if this exception had a top-level message indicating that the failures were due to the GitHub Dependency Extractor. As it stands, I suspect it won't be obvious to a user.
We could also think about the case where every resolution results in an exception, and perhaps cap the number of failures we collect/report.

Reply:
I would have added one, but the constructor for `MultipleBuildFailures` doesn't support passing a message

Reply:
Updated to use `DefaultMultiCauseException`